src/Nethermind/Nethermind.JsonRpc/Modules/Parity/ParityRpcModule.cs

```diff
@@ -77,7 +78,20 @@ public ResultWrapper<ReceiptForRpc[]> parity_getBlockReceipts(BlockParameter blo
             bool isEip1559Enabled = _specProvider.GetSpec(block.Header).IsEip1559Enabled;
             IEnumerable<ReceiptForRpc> result = receipts
                 .Zip(block.Transactions, (r, t) =>
-                    new ReceiptForRpc(t.Hash, r, t.CalculateEffectiveGasPrice(isEip1559Enabled, block.BaseFeePerGas), receipts.GetBlockLogFirstIndex(r.Index)));
+                {
+                    UInt256 effectiveGasPrice = t.CalculateEffectiveGasPrice(isEip1559Enabled, block.BaseFeePerGas);
+                    UInt256? dataGasPrice = null;
+                    ulong? dataGasUsed = null;
+                    if (t.SupportsBlobs)
+                    {
+                        dataGasPrice = IntrinsicGasCalculator.CalculateDataGasPrice(block.Header, t);
+                        dataGasUsed = IntrinsicGasCalculator.CalculateDataGas(t);
+                    }
```

Comment:
This code seems to get duplicated. Maybe lets calculate `TxGasInfo` in one call?

Reply:
You mean something like `tx.GetGasInfo()`?